libnavigation-core/src/main/java/com/mapbox/navigation/core/reroute/MapboxRerouteController.kt

```diff
@@ -123,6 +135,20 @@ internal class MapboxRerouteController @VisibleForTesting constructor(
         ) { routes, routeAlternativeId ->
             val relevantAlternative = routes.find { it.id == routeAlternativeId }
             if (relevantAlternative != null) {
+                val alternativeLegIndex = ifNonNull(
+                    tripSession.getRouteProgress(),
+                    alternativeMetadataProvider.getMetadataFor(relevantAlternative)
+                ) { routeProgress, alternativeMetadata ->
+                    val legIndex = routeProgress.currentLegProgress?.legIndex
+                    val routeGeometryIndex = routeProgress.currentRouteGeometryIndex
+                    val legGeometryIndex = routeProgress.currentLegProgress?.geometryIndex
+                    ifNonNull(legIndex, legGeometryIndex) { legIndex, legGeometryIndex ->
+                        AlternativeDataProvider.getAlternativeLegIndex(
+                            RouteProgressData(legIndex, routeGeometryIndex, legGeometryIndex),
+                            alternativeMetadata
+                        )
+                    }
+                }
```

Comment:
@mapbox/navnative , how hard it is for you to pass us leg index of alternative route together with `getLocatedAlternativeRouteId`?

Reply:
> I don't think they track alternative route progress. 

Maybe it's time to start? 🙂 
I guess iOS have the same issue with multileg routes, @mapbox/navigation-ios , do you?

Reply:
iOS use native reroute controller and refresh controller, as far as I understand.
Started a conversation in Slack.